### PR TITLE
Introduce WildFly module

### PR DESCRIPTION
**Pull request by rieckpil**
Can you have a first look @aguibert ? I wasn't sure about the license text in the `WildFlyAdapter` and if you want all changes in one commit or smaller commits. 

Furthermore what formatting options do you use? I didn't auto-format with IntelliJ as it was different to yours (maybe introduce a `.editorconfig` file?).

Signed-off-by: Philip Riecks <mail@philipriecks.de>

### CODE REVIEW

File: modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        .add(appName, "/opt/jboss/wildfly/standalone/deployments")
+                        .build())
+                .withFileFromFile(appName, appFile);
+        return image;
+
+    }
+
+    @Override
+    public Optional<String> getReadinessPath() {
+        return Optional.of("/health");
```
Comment (aguibert) [inline modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java]: Do you know if the /health endpoint is always enabled in wildfly 18.0.1.Final? Also, does it seem so be pretty consistent that the /health endpoint only returns HTTP 200 when the app is ready to serve requests?

**Comment by aguibert** [comment]
I use Eclipse as my editor, not sure if it honors .editorconfig or not. I might add a code formatting plugin which would format the code as part of the build (Quarkus uses this too)

File: modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java (hunk 1 above)
Comment (rieckpil) [reply modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java]: thanks for this hint. I checked it and in the [documentation](https://docs.wildfly.org/18/Admin_Guide.html#MicroProfile_Health_SmallRye) they say it's _included in the standalone configurations included in the WildFly distribution._ But while searching I found that the `/health` endpoint is served from the management port of WildFly (9990 by default). I think this won't work with the current setup. I could think of introducing a management port attribute within `ServerAdapter` or adjusting `getReadinessPath()` to return a `URL` ?

**Comment by rieckpil** [comment]
> I use Eclipse as my editor, not sure if it honors .editorconfig or not. I might add a code formatting plugin which would format the code as part of the build (Quarkus uses this too)

Unfortunately for Eclipse you would need a plugin (https://editorconfig.org/). If it can be automated within the build, then it's also fine :+1:

File: modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java (hunk 1 above)
Comment (aguibert) [reply modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java]: It's interesting that the documentation says the module is enabled "standalone" deployments, but then it goes on to describe how to add the module manually, which implies that there can be images of Wildfly that do not have the health module enabled?
Just to be safe lets return an empty optional here for now, perhaps we can rework the ServerAdapter readiness path later on to incorporate a port

Comment (rieckpil) [reply modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java]: Yes, I was also a little bit confused while reading the docs. I'll update the Adapter to the default behaviour in the interface

**Pull request merged**